ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlockBroadcastValidator.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Consensys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.validation;
+
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult;
+
+public class BlockBroadcastValidator {
+
+  private final BlockGossipValidator blockGossipValidator;
+
+  public BlockBroadcastValidator(final BlockGossipValidator blockGossipValidator) {
+    this.blockGossipValidator = blockGossipValidator;
+  }
+
+  public SafeFuture<BroadcastValidationResult> validate(
+      final SignedBeaconBlock block,
+      final BroadcastValidation broadcastValidation,
```

Comment:
Maybe `BroadcastValidationMode` and the same for the var name? It's not clear that it's just enum and not an action class.

Reply:
`BroadcastValidationLevel` another name

Reply:
I like `BroadcastValidationLevel` because it matches the Beacon API spec:
> broadcast_validation: Level of validation that must be applied to a block before it is broadcast.